tests/draft7/ref.json

```diff
@@ -608,5 +608,37 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "$id must be resolved against nearest parent, not just immediate parent",
+        "schema": {
+            "$id": "http://example.com/a.json",
+            "definitions": {
+                "x": {
+                    "$id": "http://example.com/b/c.json",
+                    "not": {
+                        "definitions": {
+                            "y": {
+                                "$id": "d.json",
+                                "type": "number"
+                            }
+                        }
+                    }
+                }
+            },
+            "$ref": "http://example.com/b/d.json"
```

Comment:
This `$ref` needs to be wrapped in an `allOf`.

Reply:
done wrapped in `allOf`. also added to other drafts